lib/ldap.c

```diff
@@ -183,38 +183,43 @@ const struct Curl_handler Curl_handler_ldaps = {
 #endif
 
 #ifdef USE_WIN32_LDAP
-static int Win_ldap_bind(struct connectdata *conn, LDAP *server, TCHAR *user, TCHAR *passwd)
+static int Win_ldap_bind(struct connectdata *conn,
+                         LDAP *server,
+                         TCHAR *user,
+                         TCHAR *passwd)
 {
-    int rc = 0;
-    ULONG method = LDAP_AUTH_NEGOTIATE;
-    SEC_WINNT_AUTH_IDENTITY cred = { 0 };
-    PCHAR inuser = NULL;
-    PCHAR inpass = NULL;
-
-    if (user != NULL && passwd != NULL) {
-        cred.User = user;
-        cred.UserLength = strlen(user);
-        cred.Password = passwd;
-        cred.PasswordLength = strlen(passwd);
-        cred.Flags = SEC_WINNT_AUTH_IDENTITY_ANSI;
-        if (conn->data->set.httpauth & CURLAUTH_BASIC) {
-            method = LDAP_AUTH_SIMPLE;
-            inuser = user;
-            inpass = passwd;
-        }
-        else if (conn->data->set.httpauth & CURLAUTH_DIGEST) {
-            method = LDAP_AUTH_DIGEST;
-            inpass = (TCHAR*)(&cred);
-        }
-        else if (conn->data->set.httpauth & CURLAUTH_NTLM) {
-            method = LDAP_AUTH_NTLM;
-            inpass = (TCHAR*)(&cred);
-        }
-        else {
-            inpass = (TCHAR*)(&cred);
-        }
+  ULONG method = LDAP_AUTH_NEGOTIATE;
+  SEC_WINNT_AUTH_IDENTITY cred = { 0 };
+  PCHAR inuser = NULL;
+  PCHAR inpass = NULL;
+
+  if(user != NULL && passwd != NULL) {
+    cred.User = user;
+    cred.UserLength = _tcslen(user);
+    cred.Password = passwd;
+    cred.PasswordLength = _tcslen(passwd);
+    cred.Flags = (sizeof(TCHAR) == sizeof(char)
+                ? SEC_WINNT_AUTH_IDENTITY_ANSI
+                : SEC_WINNT_AUTH_IDENTITY_UNICODE);
```

Comment:
@jay It should be good if someone will just provide .clang-format rules for this :)
Will update it shortly.